### ISSUE TITLE
[a11y] Make Notifications more accessible

### ISSUE THREAD

**Issue by unlikelyzero** [nasa/openmct#6094]
This ticket captures the work to make Open MCT notifications more accessible. This ticket identifies top level components, use cases, research, and action plan for making the current notifications meet some level of WCAG standard.

## Notification Components
We have the [notificationIndicator component](https://github.com/nasa/openmct/tree/master/src/plugins/notificationIndicator/components) which serves the following usecases:

### Notification Menu
1. Notification Menu
2. Captures all notifications of a given severity.

### Notification Alert Indicator
1. Displays alerts of all severity.
2. Can function as a hyperlink to another Domain object (Notebook Snapshot)
3. Can be configured to appear briefly before disappearing

## List of known use cases
1. 'Save Successful' to notify the user that their previous action has been saved.
It appears briefly and can be closed.
![Screen Shot 2023-01-03 at 10 16 06 AM](https://user-images.githubusercontent.com/12446869/210417168-596afe66-69b3-470b-b963-471a79114cf9.png)
2. Notebook Save with contextual hyperlink to notebook
![Screen Shot 2023-01-03 at 10 15 16 AM](https://user-images.githubusercontent.com/12446869/210419978-91814d62-4328-4f61-91f5-22f5784c5e9d.png)
3. Notification of Timespan View Changes
![Screen Shot 2023-01-03 at 10 14 54 AM](https://user-images.githubusercontent.com/12446869/210420087-ef263ec0-0cb1-4f38-a6c2-b32d6bfa5046.png)

**Comment by 360macky** [comment]
Yeah, I've been checking on it. It's a rather big point, because other components should also be updated. I will describe (and update) here what I find.

Important: The implementation of screen readers may vary. If you're using Mac, the native VoiceOver feature has issues in Chrome (it doesn't mention the alert), but it does in Safari. However, there are some more complicated ways to achieve the effect in Chrome.

I think NVDA is the most popular screen reader (but it only works for Windows)

---

Changes that aim to improve the accessibility of the notification system:

When a sudden error appears, the user flow must be interrupted to display this error. In screen-reader this did not happen.

In this case, the first change made would be to add role="alertdialog" or role="alert". The difference is that "alertdialog" is used when a user action is required (such as OK, Accept or Dismiss). So when you have the Dismiss button, I think alertdialog would be appropriate.

There are two modes of announcing the alert: "assertive" which will interrupt the user flow, and "polite" which in the event that the screen reader is reading something at the moment, will allow it to finish before announcing the notification.

We can differentiate them in this way. Error notifications will be "assertive", and warning/success will be "polite".

**Comment by unlikelyzero** [comment]
Follow this for what's next https://sheribyrnehaber.medium.com/designing-toast-messages-for-accessibility-fb610ac364be
